libs/application/api/payment/src/lib/payment.service.ts

```diff
@@ -162,58 +185,123 @@ export class PaymentService {
     applicationId: string,
     extraData: ExtraData[] | undefined,
   ): Promise<CreateChargeResult> {
-    try {
-      //.1 Get charge items from FJS
-      const chargeItems = await this.findChargeItems(
-        performingOrganizationID,
-        chargeItemCodes,
+    //1. Retrieve charge items from FJS
+    const chargeItems = await this.findChargeItems(
+      performingOrganizationID,
+      chargeItemCodes,
+    )
+
+    //2. Fetch existing payment if any
+    let paymentModel = await this.findPaymentByApplicationId(applicationId)
+    let user4 = ''
+
+    if (paymentModel) {
+      //payment Model already exists something has previously gone wrong.
+
+      const chargeStatus = await this.chargeFjsV2ClientService.getChargeStatus(
+        paymentModel.id,
       )
 
-      //2. Create and insert payment db entry
-      const paymentModel = await this.createPaymentModel(
+      if (chargeStatus === null) {
+        //No charge present in FJS - we need to create a new one
+        const chargeResult = await this.createNewCharge(
+          paymentModel,
+          user,
+          extraData,
+        )
+        user4 = chargeResult.user4
+      } else if (
+        chargeStatus.statusResult.status ===
```

Comment:
readability could be improved by having all these payment status checks in a function which would return the correct status